### PR TITLE
Sync APIs from karmada repo based on v1.5.0

### PR DESCRIPTION
**Pull request by RainbowMango**
This PR syncs APIs from karmada-io/karmada for v1.5.0.

Fixes https://github.com/karmada-io/karmada/issues/3246

### CODE REVIEW

**Comment by RainbowMango** [comment]
cc @calvin0327 could you please take a look?

File: cluster/types.go [hunk 1]
```diff
@@ -199,7 +199,7 @@ type ResourceModel struct {
 // ResourceModelRange describes the detail of each modeling quota that ranges from min to max.
 // Please pay attention, by default, the value of min can be inclusive, and the value of max cannot be inclusive.
 // E.g. in an interval, min = 2, max =10 is set, which means the interval [2,10).
-// This rule ensure that all intervals have the same meaning. If the last interval is +8,
+// This rule ensure that all intervals have the same meaning. If the last interval is +��,
```
Comment (liangyuanpeng) [inline cluster/types.go:202]: Seems like the file encoding format is break, and maybe it do not need to change?

File: cluster/v1alpha1/types.go [hunk 2]
```diff
@@ -211,7 +211,7 @@ type ResourceModel struct {
 // ResourceModelRange describes the detail of each modeling quota that ranges from min to max.
 // Please pay attention, by default, the value of min can be inclusive, and the value of max cannot be inclusive.
 // E.g. in an interval, min = 2, max =10 is set, which means the interval [2,10).
-// This rule ensure that all intervals have the same meaning. If the last interval is +8,
+// This rule ensure that all intervals have the same meaning. If the last interval is +��,
```
Comment (liangyuanpeng) [inline cluster/v1alpha1/types.go:214]: Same problem.

File: cluster/types.go (hunk 1 above)
Comment (RainbowMango) [reply cluster/types.go]: fixed both of them. Thanks.

**Comment by RainbowMango** [comment]
Please take another look.

**Review by liangyuanpeng — APPROVED** [review]

File: cluster/v1alpha1/types.go [hunk 3]
```diff
@@ -211,7 +211,7 @@ type ResourceModel struct {
 // ResourceModelRange describes the detail of each modeling quota that ranges from min to max.
 // Please pay attention, by default, the value of min can be inclusive, and the value of max cannot be inclusive.
 // E.g. in an interval, min = 2, max =10 is set, which means the interval [2,10).
-// This rule ensure that all intervals have the same meaning. If the last interval is +8,
+// This rule ensure that all intervals have the same meaning. If the last interval is +∞,
```
Comment (liangyuanpeng) [inline cluster/v1alpha1/types.go:214]: I missed it, still problem.

```
cluster/v1alpha1/types.go
cluster/types.go
```

Comment (RainbowMango) [reply cluster/v1alpha1/types.go:214]: Seems no problem here. 
But I think we can replace the `+∞` with `infinite`. 

By the way, would like to send a PR to karmada repo for the replacement?

**Comment by RainbowMango** [comment]
Force pushed, for replacing `+∞` with `infinite`. Please take another look.

**Review by liangyuanpeng — APPROVED** [review]

**Pull request merged**